metricflow/sql/sql_exprs.py

```diff
@@ -776,6 +781,148 @@ def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D
+    def requires_parenthesis(self) -> bool:  # noqa: D
+        return False
+
+    def accept(self, visitor: SqlExpressionNodeVisitor) -> VisitorOutputT:  # noqa: D
+        return visitor.visit_window_function_expr(self)
+
+    @property
+    def description(self) -> str:  # noqa: D
+        return f"{self._sql_function.value} Window Function Expression"
+
+    @property
+    def displayed_properties(self) -> List[DisplayedProperty]:  # noqa: D
+        return (
+            super().displayed_properties
+            + [DisplayedProperty("function", self.sql_function)]
+            + [DisplayedProperty("argument", x) for x in self.sql_function_args]
+            + [DisplayedProperty("partition_by_argument", x) for x in self.partition_by_args]
+            + [DisplayedProperty("order_by_argument", x) for x in self.order_by_args]
+        )
+
+    @property
+    def sql_function(self) -> SqlWindowFunction:  # noqa: D
+        return self._sql_function
+
+    @property
+    def sql_function_args(self) -> List[SqlExpressionNode]:  # noqa: D
+        return self._sql_function_args or []
+
+    @property
+    def partition_by_args(self) -> List[SqlExpressionNode]:  # noqa: D
+        return self._partition_by_args or []
+
+    @property
+    def order_by_args(self) -> List[SqlWindowOrderByArg]:  # noqa: D
+        return self._order_by_args or []
+
+    def __repr__(self) -> str:  # noqa: D
+        return f"{self.__class__.__name__}(node_id={self.node_id}, sql_function={self.sql_function.name})"
+
+    def rewrite(  # noqa: D
+        self,
+        column_replacements: Optional[SqlColumnReplacements] = None,
+        should_render_table_alias: Optional[bool] = None,
+    ) -> SqlExpressionNode:
+        return SqlWindowFunctionExpression(
+            sql_function=self.sql_function,
+            sql_function_args=[
+                x.rewrite(column_replacements, should_render_table_alias) for x in self.sql_function_args
+            ],
+            partition_by_args=[
+                x.rewrite(column_replacements, should_render_table_alias) for x in self.partition_by_args
+            ],
+            order_by_args=[
+                SqlWindowOrderByArg(
+                    expr=x.expr.rewrite(column_replacements, should_render_table_alias),
+                    descending=x.descending,
+                    nulls_last=x.nulls_last,
+                )
+                for x in self.order_by_args
+            ],
+        )
+
+    @property
+    def lineage(self) -> SqlExpressionTreeLineage:  # noqa: D
+        return SqlExpressionTreeLineage.combine(
+            tuple(x.lineage for x in self.parent_nodes) + (SqlExpressionTreeLineage(function_exprs=(self,)),)
+        )
+
+    def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D
+        if not isinstance(other, SqlWindowFunctionExpression):
+            return False
+        order_by_matches = all(x == y for x, y in itertools.zip_longest(self.order_by_args, other.order_by_args))
```

Comment:
updated the check to do `len(x) == len(y) and set(x) == set(y)`.

Reply:
nvm, we should actually just do an equality check since ordering matters for this. So I just made it a simple `self.order_by_args == other.order_by_args`